src/plugins/workspace/public/components/workspace_detail/opensearch_connections_table.tsx

```diff
@@ -3,60 +3,114 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import React, { useMemo, useState } from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import {
   EuiSpacer,
-  EuiButton,
-  EuiFlexItem,
-  EuiFlexGroup,
-  EuiFieldSearch,
   EuiInMemoryTable,
   EuiBasicTableColumn,
   EuiTableSelectionType,
   EuiTableActionsColumnType,
   EuiConfirmModal,
+  EuiSearchBarProps,
+  EuiText,
+  EuiListGroup,
+  EuiListGroupItem,
+  EuiIcon,
+  EuiPopover,
+  EuiButtonEmpty,
+  EuiPopoverTitle,
+  EuiSmallButton,
 } from '@elastic/eui';
 import { i18n } from '@osd/i18n';
 import { CoreStart, WorkspaceObject } from '../../../../../core/public';
-import { DataSource } from '../../../common/types';
+import { DataSource, DataSourceConnection, DataSourceConnectionType } from '../../../common/types';
 import { WorkspaceClient } from '../../workspace_client';
 import { convertPermissionSettingsToPermissions, useWorkspaceFormContext } from '../workspace_form';
 import { useOpenSearchDashboards } from '../../../../opensearch_dashboards_react/public';
+import PrometheusLogo from '../../assets/prometheus_logo.svg';
+import S3Logo from '../../assets/s3_logo.svg';
 
 interface OpenSearchConnectionTableProps {
-  assignedDataSources: DataSource[];
   isDashboardAdmin: boolean;
   currentWorkspace: WorkspaceObject;
   setIsLoading: React.Dispatch<React.SetStateAction<boolean>>;
+  connectionType: string;
```

Comment:
We can not use DataSourceConnectionType, because the passed parameter is toggleButton id which must be the type of string.
```
{
      id: 'openSearchConnections',
      label: i18n.translate('workspace.detail.dataSources.openSearchConnections', {
        defaultMessage: 'OpenSearch connections',
      }),
    },
```